beacon-chain/sync/initial-sync/blocks_fetcher_utils.go

```diff
@@ -226,37 +233,51 @@ func (f *blocksFetcher) findForkWithPeer(ctx context.Context, pid peer.ID, slot
 	// Request blocks starting from the first non-empty slot.
 	req := &p2ppb.BeaconBlocksByRangeRequest{
 		StartSlot: nonSkippedSlot,
-		Count:     uint64(slotsPerEpoch.Mul(2)),
+		Count:     reqCount,
 		Step:      1,
 	}
 	blocks, err := f.requestBlocks(ctx, req, pid)
 	if err != nil {
 		return nil, fmt.Errorf("cannot fetch blocks: %w", err)
 	}
+	if len(blocks) == 0 {
+		return nil, errNoAlternateBlocks
+	}
+	first := blocks[0]
+	if !f.chain.HasBlock(ctx, first.Block().ParentRoot()) {
```

Comment:
Why do we have this here ?

Reply:
discussed in live review, will add comments

Reply:
added